src/battle_util.c

```diff
@@ -3412,6 +3446,95 @@ u8 AtkCanceller_UnableToUseMove(void)
             }
             gBattleStruct->atkCancellerTracker++;
             break;
+        case CANCELLER_MULTIHIT_MOVES:
+            if (gBattleMoves[gCurrentMove].effect == EFFECT_MULTI_HIT)
+            {
+                u16 ability = gBattleMons[gBattlerAttacker].ability;
+
+                if (ability == ABILITY_SKILL_LINK)
+                {
+                    gMultiHitCounter = 5;
+                }
+                #ifdef POKEMON_EXPANSION
+                else if (ability == ABILITY_BATTLE_BOND
+                && gCurrentMove == MOVE_WATER_SHURIKEN
+                && gBattleMons[gBattlerAttacker].species == SPECIES_GRENINJA_ASH)
+                {
+                    gMultiHitCounter = 3;
+                }
+                #endif
+                else
+                {
+                    if (B_MULTI_HIT_CHANCE >= GEN_5)
+                    {
+                        // 2 and 3 hits: 33.3%
+                        // 4 and 5 hits: 16.7%
+                        gMultiHitCounter = Random() % 4;
+                        if (gMultiHitCounter > 2)
+                        {
+                            gMultiHitCounter = (Random() % 3);
+                            if (gMultiHitCounter < 2)
+                                gMultiHitCounter = 2;
+                            else
+                                gMultiHitCounter = 3;
+                        }
+                        else
+                            gMultiHitCounter += 3;
+                    }
+                    else
+                    {
+                        // 2 and 3 hits: 37.5%
+                        // 4 and 5 hits: 12.5%
+                        gMultiHitCounter = Random() % 4;
+                        if (gMultiHitCounter > 1)
+                            gMultiHitCounter = (Random() % 4) + 2;
+                        else
+                            gMultiHitCounter += 2;
+                    }
+                }
+
+                PREPARE_BYTE_NUMBER_BUFFER(gBattleScripting.multihitString, 1, 0)
+            }
+            else if (IsTwoStrikesMove(gCurrentMove))
+            {
+                gMultiHitCounter = 2;
+				PREPARE_BYTE_NUMBER_BUFFER(gBattleScripting.multihitString, 1, 0)
```

Comment:
Spaces instead of tabs. Also please align the value setting of `gMultiHitCounter` with the call to the `PREPARE_BYTE_NUMBER_BUFFER` macro.